### PR TITLE
Adding User Profile 

### PR DESCRIPTION
**Pull request by SubhamB2003**
Issue Solved: #49

### CODE REVIEW

**Comment by SubhamB2003** [comment]
@narayan954 Hey, can I pull the previous repo and after making the following changes can I create a new PR?

**Comment by narayan954** [comment]
> @narayan954 Hey, can I pull the previous repo and after making the following changes can I create a new PR?

Hi, it's okay, I'll rebase it. You can just run git pull -f to update your branch

**Comment by narayan954** [comment]
> @narayan954 Hey, can I pull the previous repo and after making the following changes can I create a new PR?

I didn't clearly understand. Did you mean you want to work on other issues as well? Well for that, first update your fork's main branch with mine , then git checkout to main, create other branch for the feature from there.

**Comment by SubhamB2003** [comment]
> > @narayan954 Hey, can I pull the previous repo and after making the following changes can I create a new PR?
> 
> I didn't clearly understand. Did you mean you want to work on other issues as well? Well for that, first update your fork's main branch with mine , then git checkout to main, create other branch for the feature from there.

Okk got it.

I added a user feature in this PR can you merge it?
You can see this here
Previous:
image

After Adding:
image

Profile Page
image

**Comment by narayan954** [comment]
> > > @narayan954 Hey, can I pull the previous repo and after making the following changes can I create a new PR?
> > 
> > 
> > I didn't clearly understand. Did you mean you want to work on other issues as well? Well for that, first update your fork's main branch with mine , then git checkout to main, create other branch for the feature from there.
> 
> Okk got it.
> 
> I added a user feature in this PR can you merge it? You can see this here Previous: image
> 
> After Adding: image
> 
> Profile Page image

Hi @SubhamB2003 , this looks fine, I am awaiting for a mentor.
Can you add this link to profile on the profile icon of the posts as well, so that anyone can see profile of the person by clicking on the profile icon. (later this would be used to add followers and profile feed as well).

I am not asking this for this pr,only but as a new feature request after this feature is implemented.

**Comment by SubhamB2003** [comment]
@narayan954 Hey, Are you saying this to implement this feature right?

Here it is

https://github.com/narayan954/dummygram/assets/94003231/44d68b67-5c4e-4a1f-8381-7c790b6763d0

**Comment by narayan954** [comment]
> @narayan954 Hey, Are you saying this to implement this feature right?
> 
> Here it is
> 
> https://github.com/narayan954/dummygram/assets/94003231/44d68b67-5c4e-4a1f-8381-7c790b6763d0
> 
> 

yes exactly!!

**Comment by SubhamB2003** [comment]
> > @narayan954 Hey, Are you saying this to implement this feature right?
> > Here it is
> > 
> >   
> >     
> >     
> > 
> >     Profile.Viewer.mp4
> 
> yes exactly!!

So can I create a feature improvement Issue and create a PR?

File: src/App.jsx [hunk 1]
```diff
@@ -50,6 +60,8 @@ function App() {
 
   const [posts, setPosts] = useState([]);
   const [user, setUser] = useState(null);
+  const [open, setOpen] = useState(false);
+  const [curUser, setCurUser] = useState([]);
```
Comment (narayan954) [inline src/App.jsx:64]: is curUser really needed to shadow the user? I don't think we need a curUser when user and curUser both are same entity. About using to json, you may use it on the user variable or while passing it as prop to the profile route.

TLDR: We don't need unnecessary variables/useStates.

**Review by narayan954 — CHANGES_REQUESTED** [review]
minor change, rest all good!

**Comment by narayan954** [comment]
> > > @narayan954 Hey, Are you saying this to implement this feature right?
> > > Here it is
> > > ```
> > > Profile.Viewer.mp4
> > > ```
> > 
> > 
> > yes exactly!!
> 
> So can I create a feature improvement Issue and create a PR?

yeah sure!

File: src/App.jsx (hunk 1 above)
Comment (SubhamB2003) [reply src/App.jsx:64]: Yes, you are right it's my fault. I create a new PR that uses the existing user state.
 
You can now review it.
